src/Chapter2.hs

```diff
@@ -842,7 +867,16 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n l
+  | n < 0 || null l = []
+  | otherwise = drop (n `mod` length l) l ++ take (n `mod` length l) l
```

Comment:
Nice!
You can move 
```haskell
n `mod` length l
```
into a separate var and use it everywhere.
Also, using `cycle` could be a bit faster here as you won't `++` two lists in that case 🙂